spylib/token.py

```diff
@@ -282,3 +300,45 @@ def load(cls, store_name: str, associated_user: str):
         This method handles loading the token. By default this does nothing,
         therefore the developer should override this.
         """
+
+
+async def get_token(
+    domain: str,
+    code: str,
+    api_key: str,
+    api_secret_key: str,
+    CustomToken: Union[Type[OfflineTokenABC], Type[OnlineTokenABC]],
+    Response: Type[TokenResponse],
+) -> Union[OnlineTokenABC, OfflineTokenABC]:
+    """
+    Get token from shopify. Accepts a Token subclass, which will be the type of token
+    that is generated, and the TokenResponse type which should match the Token subclass
+    type.
+    """
+
+    url = f'https://{domain}/admin/oauth/access_token'
+
+    httpclient = HTTPClient()
+
+    jsondata = {'client_id': api_key, 'client_secret': api_secret_key, 'code': code}
+    response = await httpclient.request(
+        method='post',
+        url=url,
+        json=jsondata,
+        timeout=20.0,
+    )
+    if response.status_code != 200:
+        message = (
+            f'Problem retrieving access token. Status code: {response.status_code} {jsondata}'
+            f'response=> {response.text}'
+        )
+        logger.error(message)
+        raise ValueError(message)
+
+    jresp = Response.parse_obj(response.json())
+
+    # We have to use this syntax
+    return CustomToken(
+        **{'store_name': domain_to_storename(domain)},
+        **jresp.dict(),
+    )
```

Comment:
This merges the dict and the store name value. Because the response from the API misses the store name.